lib/ddtrace/contrib/sidekiq/tracing.rb

```diff
@@ -18,9 +20,24 @@ def initialize(options = {})
         def job_resource(job)
           if job['wrapped']
             job['wrapped']
+          elsif job['class'] == 'Sidekiq::Extensions::DelayedClass'
+            delay_extension_class(job)
           else
             job['class']
           end
+        rescue => e
+          Datadog::Logger.log.debug("Error retrieving Sidekiq job class name (jid:#{job['jid']}): #{e}")
+
+          job['class']
+        end
+
+        #
+        def delay_extension_class(job)
+          clazz, method = YAML.parse(job['args'].first).children.first.children
```

Comment:
If `parse` is safe, I think that will address the security concerns from the previous version of this.

Reply:
👍